pvfactors/tests/test_geometry/test_pvarray.py

```diff
@@ -497,3 +497,16 @@ def test_time_ordered_pvarray(params):
         list_elapsed.append(toc - tic)
 
     print("\nAvg time elapsed: {} s".format(np.mean(list_elapsed)))
+
+def test_ordered_pvarray_gnd_shadow_casting_tolerance():
+    params = {'axis_azimuth': 0, 
+            'gcr': 0.3,
+            'n_pvrows': 3,
+            'pvrow_height': 1.8,
+            'pvrow_width': 1.98,
+            'solar_azimuth': 263.99310644558074,
+            'solar_zenith': 73.91658668648401, 
+            'surface_azimuth': 270.0, 
+            'surface_tilt': 51.98206680806641}
+    pvarray_w_direct_shading = OrderedPVArray.from_dict(params)
+    pvarray_w_direct_shading.cast_shadows()
```

Comment:
Thanks a lot for writing this test @tcapelle !
One question: I ran this test using your branch on my laptop with and without the added ``DISTANCE_TOLERANCE`` in the ``cast_shadows()`` function you updated, and it passed in both cases... Are you sure that's the case that failed during your simulation? Could you give it another look please?

Reply:
Look my answer above.

Reply:
Hey @tcapelle sorry for the late reply. I'm still unable to reproduce this error locally, and I'm starting to think that it could be a computer roundoff kind of thing, so I'm going to merge since it doesn't seem to hurt anything!
Thanks for the contribution!